attestation-service/attestation-service/src/rvps/builtin.rs

```diff
@@ -9,16 +10,21 @@ pub struct Rvps {
 }
 
 impl Rvps {
-    pub fn new(store_type: &str) -> Result<Self> {
-        let core = Core::new(store_type)?;
-        Ok(Self { core })
+    pub fn new(store_type: &str) -> Result<Self, RvpsError> {
+        Core::new(store_type)
+            .map(|core| Self::from_core(core))
+            .map_err(|error| RvpsError::CreateRvps(error.to_string()))
+    }
+
+    fn from_core(core: Core) -> Self {
+        Self { core }
```

Comment:
why create a from_core() fn, we shouldn't  need to perform code changes if we adjust the error types, it should mostly propagate fine with `..?;`, no need for `map_err()` in most cases.

I don't think we should convert anyhow errors (`Core::new()` returns an anyhow error) and wrap them in ours. If we want to do not start "at the bottom", but do this later we can add:

```
    #[error("Error")]
    Anyhow(#[from] anyhow::Error),
```

to the error enum and the code should keep working.